### DISCUSSION TITLE
Is there a Mastodon data model document or design?

### DISCUSSION THREAD

**Discussion by chetmurthy** [mastodon/mastodon#22819] [Q&A]
I'm looking around for a data model -- something a little higher-level than the database schema exportable from a relational database, but much like that -- for Mastodon.  Does such a thing exist ?

**Comment by mjankowski** [comment]
I don't think there is anything official along these lines, but there are various tools/rubygems out there which will generate a UML diagram or other similar model from the schema and/or the model files.
